RHIDP-1921 Document Known issues and fixed issues in RHDH 1.2 release… (#308)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/con-relnotes-fixed-issues.adoc

In this release, the security requirement for dynamic plugin static assets has been removed, which restores access to dynamic plugin UI elements.

> Following the implementation of this fix, users can view and interact with dynamic plugin UI elements, resulting in improved usability and functionality within the application. 

Reviewer: ```suggestion
With this fix, users can view and interact with dynamic plugin UI elements, resulting in improved usability and functionality within the application. 
```
Make consistent with following bugs maybe?

This issue led to difficulties in defining conditional policies, hindering the configuration of permissions within the application.

> With this fix, you can now define conditional policies in {product-short} with permission policy actions of 'use'. 

Reviewer: I get that in the code this is 'use', but if it's a literal name of a field or something, should it be `use` in our docs maybe? I could be very wrong.

//link:{LinkRHIDPIssue}RHIDP-2237[RHIDP-2237]
--

Setting a custom route not working when using the {product-short} Operator on OpenShift::
+
--
In earlier versions of the {product} Operator, setting a custom Route host on OpenShift Container Platform by using the `spec.application.route.host` field in the Custom Resource was not possible.

> This limitation prevented users from configuring custom Route hosts, limiting their ability to customize the deployment environment.

Reviewer: ```suggestion
This limitation prevented users from configuring custom route hosts, limiting their ability to customize the deployment environment.
```
Similar to previous convos with Lindsey, I'd recommend using lowercase for general mentions of things like routes, and only capitalizing if we're referring to a specific "`Route` object", basically consistent with https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

> With this release, you can now set a custom Route host on OpenShift Container Platform by using the specified field in the Custom Resource. 

Reviewer: ```suggestion
With this release, you can now set a custom route host on {ocp-short} by using the specified field in the Custom Resource.
```
Normally I'd say use custom resource (CR) and then CR for each instance after the first in a module, but I see this is used in other places, so more of a note for future updates :)

> With this release, you can now set a custom Route host on OpenShift Container Platform by using the specified field in the Custom Resource. 

Reviewer 2: since "custom Route" and "Custom Resource" could both be acronym'd to "CR", are we sure we're consistently using "CR" to only refer to a Custom Resource?

> With this release, you can now set a custom Route host on OpenShift Container Platform by using the specified field in the Custom Resource. 

Author: To avoid confusion, let's go with lowercase _custom resource_.

This update adds CPU and memory requests to the default configuration of the Operator, ensuring that {product-short} pods have the necessary resource requests specified.

With this fix, {product-short} pods can now be scheduled and started properly on all clusters, ensuring the availability of the {product} application. 
//link:{LinkRHIDPIssue}RHIDP-2159[RHIDP-2159]
--

> Allow specifying image pull secrets in Custom Resource for the database::

Author: Ok

This update fixes the issue by propagating the image pull secrets specified in the `spec.application.imagePullSecrets` Custom Resource field. You can now use these secrets can for both the {product-short} and database images.

> With this fix, users can successfully use image pull secrets for both the {product-short} and database images, allowing deployments from private repositories such as `registry.redhat.io` in non-OpenShift clusters like Amazon EKS or Azure AKS. This ensures greater deployment flexibility and compatibility across different environments. 

Reviewer 3: @abrennan89  Do you know if there is any guidance on using the term "non-OpenShift"?

> With this fix, users can successfully use image pull secrets for both the {product-short} and database images, allowing deployments from private repositories such as `registry.redhat.io` in non-OpenShift clusters like Amazon EKS or Azure AKS. This ensures greater deployment flexibility and compatibility across different environments. 

Reviewer: Had the same thought :) not that I'm aware of, though

> With this fix, users can successfully use image pull secrets for both the {product-short} and database images, allowing deployments from private repositories such as `registry.redhat.io` in non-OpenShift clusters like Amazon EKS or Azure AKS. This ensures greater deployment flexibility and compatibility across different environments. 

Reviewer 2: "private repositories such as `registry.redhat.io`"

Um... registry.redhat.io isn't a private repo. 

Would "Kubernetes" be a better term than "non-OpenShift" ?

> With this fix, users can successfully use image pull secrets for both the {product-short} and database images, allowing deployments from private repositories such as `registry.redhat.io` in non-OpenShift clusters like Amazon EKS or Azure AKS. This ensures greater deployment flexibility and compatibility across different environments. 

Author: Let's go with Kubernetes.

This update ensures that the RBAC tab is hidden when the RBAC plugin is disabled.

With this fix, the RBAC tab is no longer visible when the RBAC plugin is disabled, resulting in a cleaner and more intuitive user interface. 
//link:{LinkRHIDPIssue}RHIDP-1449[RHIDP-1449]
--

Show configure access button for previously created simple permission policies in edit form::
+
--

> In previous versions, when a user creates simple permission policies and later returns to edit the role, the *Configure Access* button does not appear if they wish to add conditional permission policies.

Reviewer: ```suggestion
In previous versions, when a user creates simple permission policies and later returns to edit the role, the *Configure Access* button does not appear if they want to add conditional permission policies.
```

Reviewer 3: Yes to _want_ instead of _wish_ as a general rule. The other option is to end the sentence after _appear_, especially since the next sentence states that this issue affects being able to add conditional permission policies.

> In previous versions, when a user creates simple permission policies and later returns to edit the role, the *Configure Access* button does not appear if they wish to add conditional permission policies.

Reviewer 2: ```suggestion
In previous versions, when a user creates simple permission policies and later returns to edit the role, the *Configure Access* button does not appear.
```

As a result, the user cannot add conditional permission policies to roles that were created with simple permission policies, limiting the ability to update and refine access controls.

In this release, the role form has been updated to display the *Configure Access* button for previously created simple permission policies for plugins and resource types that support conditions. This fix allows users to add and save new conditional policies. 
//link:{LinkRHIDPIssue}RHIDP-2346[RHIDP-2346]
--

Conflicting Condition Action Sets::
+
--

> In previous versions, the Condition API allowed storing multiple conditions with conflicting action sets.

Reviewer 2: presumably anyone hitting the API could achieve this, be it a bot, script, process, crontab, or human.

//link:{LinkRHIDPIssue}RHIDP-1506[RHIDP-1506]
--

Existing Backstage operand not upgraded after upgrading operator from 1.1.x to 1.2.x::
+
--
In previous versions, there was an issue with the {product-short} Operator upgrade process that prevented operator-backed {product-short} instances from upgrading seamlessly when the {product-short} Operator itself was upgraded. This occurred because the Operator, when trying to reconcile already-existing {product-short} Custom Resources, was denied patching certain fields that are restricted or read-only by Kubernetes or OpenShift.

This issue led to a failure in reaching the desired state during the upgrade. 

In this update, the Operator has been refactored to overcome these issues by replacing objects forcibly when it is unable to patch them. However, as a known issue, users might need to re-create any custom labels or annotations on the underlying resources managed by the {product-short} Operator after the upgrade. 
//link:{LinkRHIDPIssue}RHIDP-2597[RHIDP-2597]
--

// [TBC] Failed to List Cluster Resources in Janus IDP Backstage Plugin OCM Backend Dynamic::
// +
// --

> // In previous versions, the OCM Plugin Readme file had no information on how to configure OCM on a Kubernetes (K8s) cluster.

Reviewer: I'd leave k8s out honestly, it doesn't add any value here

> // In previous versions, the OCM Plugin Readme file had no information on how to configure OCM on a Kubernetes (K8s) cluster.

Reviewer: Maybe add an attribute for this?

// Due to this missing information, users were unable to configure the OCM plugin to fetch clusters, resulting in the plugin's inability to display clusters.

> // In this release, the Readme file has been updated to include a link for configuring OCM on a K8s cluster. Additionally, we added steps to enable access to the OCM backend plugin when the RBAC permission framework is enabled.

Reviewer: I didn't review stuff that was commented out, sorry

// With these updates, users can now properly configure the OCM plugin to fetch and display clusters in the OCM front-end, ensuring the plugin operates as intended. 
//link:{LinkRHIDPIssue}RHIDP-2240[RHIDP-2240]
// --

== Fixed security issues

This section lists fixed security issues with {product} {product-version}:

link:https://access.redhat.com/security/cve/CVE-2024-35195[CVE-2024-35195]::

> An incorrect control flow implementation vulnerability was found in Requests. If the first request in a session is made with `verify=False`, all subsequent requests to the same host will continue to ignore cert verification. 

Reviewer: ```suggestion
An incorrect control flow implementation vulnerability was found in Requests. If the first request in a session is made with `verify=False`, all subsequent requests to the same host ignore cert verification. 
```
Suggest remove fluff

Reviewer 2: do we need to spell out `certificate` instead of jusg `cert` ? 
```suggestion
An incorrect control flow implementation vulnerability was found in Requests. If the first request in a session is made with `verify=False`, all subsequent requests to the same host will ignore certificate verification. 
```

//link:{LinkRHIDPIssue}RHIDP-1343[RHIDP-1343]

link:https://access.redhat.com/security/cve/CVE-2024-27307[CVE-2024-27307]::

> A vulnerability was found in JSONata. A malicious expression can exploit the transform operator to override properties on the Object constructor and prototype. This issue can result in denial of service, remote code execution, or other unforeseen behavior in applications that assess user-provided JSONata expressions. 

Reviewer 2: ```suggestion
A vulnerability was found that could exploit the JSONata transform operator to override properties on the Object constructor and prototype. This could result in denial of service, remote code execution, or other unforeseen behavior in applications that assess user-provided JSONata expressions. 
```

> A vulnerability was found in JSONata. A malicious expression can exploit the transform operator to override properties on the Object constructor and prototype. This issue can result in denial of service, remote code execution, or other unforeseen behavior in applications that assess user-provided JSONata expressions. 

Reviewer: I would say yes, but it's not clear to me if the "transform operator" is the name of an Operator or not. If this is just referring to a component, I'd try to use a less overloaded term than operator to describe it.

> A vulnerability was found in JSONata. A malicious expression can exploit the transform operator to override properties on the Object constructor and prototype. This issue can result in denial of service, remote code execution, or other unforeseen behavior in applications that assess user-provided JSONata expressions. 

Reviewer 2: See https://issues.redhat.com/browse/RHIDP-1524... I don't think this refers to an OpenShift Operator, but rather a bit of code they've chosen to call an operator". https://docs.jsonata.org/other-operators#-------transform 

"The object transform operator is used to modify a copy of an object structure using a pattern/action syntax to target specific modifications while keeping the rest of the structure unchanged."
